### PR TITLE
Upgraded to webpack 4

### PR DESCRIPTION
**Pull request by sroy3**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,10 +20,12 @@
     "url": "git+https://github.com/coveo/react-vapor.git"
   },
   "scripts": {
-    "start": "gulp docs && webpack-dev-server --devtool eval --inline --hot --progress --colors",
-    "docs": "gulp docs",
+    "start": "npm run docs && webpack-dev-server --mode development --config webpack.config.js --devtool eval --inline --hot --progress --colors",
+    "docs": "webpack --config webpack.config.js",
```
Comment (MichelML) [inline package.json]: `webpack` implique qu'on aille webpack 4 installer globalement sur notre ordi, je propose d'essayer d'utiliser `npx webpack --config webpack.config.js` https://www.npmjs.com/package/npx

Comment (MichelML) [reply package.json]: ou `node node_modules/pathToWebpackCommand`

**Comment by MichelML** [comment]
tout semble beau dans le demo https://coveo.github.io/react-vapor/upgrade-webpack/

File: package.json (hunk 1 above)
Comment (sroy3) [reply package.json]: I'll check, but I did not upgrade my local version of webpack for it to be updated, simply upgrade the version in package.json. What you're saying makes sense, but I'm wondering why it works anyway.

Comment (MichelML) [reply package.json]: You are right! TIL https://docs.npmjs.com/cli/run-script#description --> "Any binaries provided by locally-installed dependencies can be used without the node_modules/.bin prefix." This is very good to know :)

File: webpack.config.prod.minify.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const prodConfig = require('./webpack.config.prod');
+
+prodConfig.output.filename = 'react-vapor.min.js';
+prodConfig.optimization = {};
+prodConfig.optimization.minimize = true;
```
Comment (GermainBergeron) [inline webpack.config.prod.minify.js]:
```
prodConfig.output.filename = 'react-vapor.min.js';
prodConfig.optimization = {minimize: true};
```

**Pull request merged**